pyresample/gradient/__init__.py

```diff
@@ -306,10 +316,9 @@ def _gradient_resample_data(src_data, src_x, src_y,
     assert src_gradient_yp.ndim == 2
     assert dst_x.ndim == 2
     assert dst_y.ndim == 2
-    all_shapes = (src_data.shape[1:], src_x.shape, src_y.shape,
-                  src_gradient_xl.shape, src_gradient_xp.shape,
-                  src_gradient_yl.shape, src_gradient_yp.shape)
-    assert all(all_shapes[0] == shape for shape in all_shapes[1:])
+    assert (src_data.shape[1:] == src_x.shape == src_y.shape ==
+            src_gradient_xl.shape == src_gradient_xp.shape ==
+            src_gradient_yl.shape == src_gradient_yp.shape)
```

Comment:
Most code checkers don't like asserts outside of test functions. I've started to agree with them. Could these be made into explicit checks with ValueErrors being raised?

Reply:
Fixed